src/propulsion/calculate_RPM.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Author: George Karabassis
+ * Organisation: HYPED
+ * Date: 30/3/2018
+ * Description:
+ *
+ *    Copyright 2018 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ *    except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *    either express or implied. See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "propulsion/calculate_RPM.hpp"
+
+#include <math.h>
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+#include <cstdint>
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <algorithm>
+
+// #include "motor_control/communicator.hpp"
+#include "data/data.hpp"
+#include "utils/system.hpp"
+
+namespace hyped
+{
+
+using data::NavigationType;
+using utils::System;
+using utils::Timer;
+
+namespace motor_control
+{
+
+constexpr double kHalbachRadius = 0.148;
+const std::string kAccelerationData = "../BeagleBone_black/data/configuration/AccelerationSlip.txt";
+
+CalculateRPM::CalculateRPM(Logger &log) : data_(data::Data::getInstance()),
+                                          acceleration_slip_(),
+                                          log_(log)
+{
+}
+
+bool CalculateRPM::initialize(std::string filepath)
+{
+    return check_file(filepath);
```

Comment:
after a check to check_file here, it might be good to call RPM_calc here with the given filepath

Reply:
Updated it.